src/machine/usb/cdc/usbcdc.go

```diff
@@ -0,0 +1,193 @@
+package cdc
+
+import (
+	"errors"
+	"machine"
+	"runtime/interrupt"
+)
+
+var (
+	ErrBufferEmpty = errors.New("USB-CDC buffer empty")
+)
+
+const cdcLineInfoSize = 7
+
+type cdcLineInfo struct {
+	dwDTERate   uint32
+	bCharFormat uint8
+	bParityType uint8
+	bDataBits   uint8
+	lineState   uint8
+}
+
+// Read from the RX buffer.
+func (usbcdc *USBCDC) Read(data []byte) (n int, err error) {
+	// check if RX buffer is empty
+	size := usbcdc.Buffered()
+	if size == 0 {
+		return 0, nil
+	}
+
+	// Make sure we do not read more from buffer than the data slice can hold.
+	if len(data) < size {
+		size = len(data)
+	}
+
+	// only read number of bytes used from buffer
+	for i := 0; i < size; i++ {
+		v, _ := usbcdc.ReadByte()
+		data[i] = v
+	}
+
+	return size, nil
+}
+
+// ReadByte reads a single byte from the RX buffer.
+// If there is no data in the buffer, returns an error.
+func (usbcdc *USBCDC) ReadByte() (byte, error) {
+	// check if RX buffer is empty
+	buf, ok := usbcdc.rxBuffer.Get()
+	if !ok {
+		return 0, ErrBufferEmpty
+	}
+	return buf, nil
+}
+
+// Buffered returns the number of bytes currently stored in the RX buffer.
+func (usbcdc *USBCDC) Buffered() int {
+	return int(usbcdc.rxBuffer.Used())
+}
+
+// Receive handles adding data to the UART's data buffer.
+// Usually called by the IRQ handler for a machine.
+func (usbcdc *USBCDC) Receive(data byte) {
+	usbcdc.rxBuffer.Put(data)
+}
+
+// USBCDC is the USB CDC aka serial over USB interface.
+type USBCDC struct {
+	rxBuffer *rxRingBuffer
+	txBuffer *txRingBuffer
+	waitTxc  bool
+}
+
+func (x *USBCDC) Debug() int {
+	return 3
```

Comment:
3? What does that number mean in this context?

Reply:
I forgot to delete the code for debugging.

Reply:
:smiley_cat: